internal/component/database_observability/mysql/collector/setup_consumers.go

```diff
@@ -0,0 +1,141 @@
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/atomic"
+
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+)
+
+const (
+	SetupConsumersName = "setup_consumers"
+)
+
+type SetupConsumerArguments struct {
+	DB       *sql.DB
+	Registry *prometheus.Registry
+
+	Logger         log.Logger
+	ScrapeInterval time.Duration
+}
+
+type SetupConsumers struct {
+	dbConnection         *sql.DB
+	Registry             *prometheus.Registry
+	collectInterval      time.Duration
+	SetupConsumersMetric *prometheus.GaugeVec
```

Comment:
I think all fields here can be kept private? (I'm referring to `Registry` and `SetupConsumersMetric`)